doc/sphinx/electrostatics.rst

```diff
@@ -4,19 +4,21 @@ Electrostatics
 ==============
 
 The Coulomb (or electrostatic) interaction is defined as
-follows. For a pair of particles at distance :math:`r` with charges
-:math:`q_1` and :math:`q_2`, the interaction is given by
+follows. For a pair of particles at distance :math:`r` with formal charges
+:math:`z_1` and :math:`z_2`, the electric potential energy is given by
 
-.. math:: U_C(r)=C \cdot \frac{q_1 q_2}{r}.
+.. math:: U_C(r)=C \cdot \frac{z_1 z_2}{r}.
 
 where
 
 .. math::
-   C=\frac{1}{4\pi \varepsilon_0 \varepsilon_r}
+   C=\frac{e^2}{4\pi \varepsilon_0 \varepsilon_r}
```

Comment:
I don't believe this to be correct. I don't think that any of this is related to the elementary charge (which is a constant of nature, one absolute amount of charge, whose numerical value depends on the unit system). I rather think that the charges are measured in whatever the charge units are.

Reply:
We're missing an `e^2` somewhere to make these equations consistent. The issue here is about definition. You probably think of the Coulomb prefactor `C` as the Coulomb constant `k_e = 1/ (4 * pi * e0)`, which is also what I intuitively assumed the first time I read the text, but the text actually defines `C` with regards to the Bjerrum length.

Reply:
Well the interaction energy in espresso is `C * q_1 * q_2 / r`, independently of what the text defines...

Reply:
We could rewrite the text, starting from the Coulomb equation `U_C = k_e * q1 * q2 / r` with `k_e = 1/ (4 * pi * e0)` and then introduce the ESPResSo convention where `U_C = C * z1 * z2 / r` with `C` defined from the Bjerrum length. Most ESPResSo users prefer to put an integer charge (aka charge number) on their coarse-grained particles and delegate the actual units to the prefactor `C`. This would clarify the ESPResSo convention while remaining in line with physics textbooks.

Reply:
I find it odd to push a unit system here, frankly. And charge numbers are not a concept that exists in electrodynamics. If you particles happen do be not ions, `e` is by no means a natural charge unit.

Reply:
Then we either write `C = l_B * k_BT / e^2`, or we drop the sentence about the Bjerrum length.

Reply:
I would do the former but this is also not a hill I'm willing to die on. I just find the idea that quantities have specific natural unit odd and mislead in general, and confusing in Espresso in particular because it does not have a fix unit system.

Reply:
> `e` is by no means a natural charge unit

This is a valid argument, how about introducing z_i as "normalized charges" `z_i = q_i / e` instead of formal charges?

Reply:
I'm not clear still on why they have to be introduced at all.

Reply:
Why is there da "definition" for `C` at all, when right after it says that is a settable parameter?
So I'd do the following. Drop the definition of `C` as such, then two examples which `C` to use for a relative permittivity, and for a specific Bjerrum length?

Reply:
It would be simpler to just rewrite `C=l_B k_B T / e^2` to fix the inconsistency. As you said above and in https://github.com/espressomd/espresso/pull/3673#issuecomment-616797265, the user can move `q` around for convenience in simulations with ions, without changing the energy.